test/e2e/reporter.go

```diff
@@ -123,21 +184,32 @@ func (r *KubernetesNMStateReporter) logPods(testName string, namespace string, s
 			io.WriteString(writer, formattedLogs)
 		}
 	})
+
+	// Let's print the pods logs to the GinkgoWriter so
+	// we see the failure directly at prow junit output without opening files
+	if podsLogFile != "" {
```

Comment:
Why to read file if you already have those logs in formattedLogs?

Reply:
Ahh you right I was just lazy, let's make func(io.Writer) be func(io.Writer) string and return it so it's used elsewhere